internal/pkg/manifest/validate.go

```diff
@@ -1933,40 +1955,58 @@ func populateALBPortsAndValidate(containerNameFor map[uint16]string, opts valida
 }
 
 func populateNLBPortsAndValidate(containerNameFor map[uint16]string, opts validateExposedPortsOpts) error {
-	if opts.nlb == nil {
+	if opts.nlb == nil || opts.nlb.MainListener.IsEmpty() {
 		return nil
 	}
 	nlb := opts.nlb
-	if nlb.Port == nil {
-		return nil
+	if err := populateAndValidateNLBPorts(nlb.MainListener, containerNameFor, opts.mainContainerName); err != nil {
+		return fmt.Errorf(`validate nlb.listener: %w`, err)
 	}
-	nlbPort, _, err := ParsePortMapping(nlb.Port)
+
+	for idx, listener := range nlb.AdditionalListeners {
+		if err := populateAndValidateNLBPorts(listener, containerNameFor, opts.mainContainerName); err != nil {
+			return fmt.Errorf(`validate nlb.listener[%d]: %w`, idx, err)
```

Comment:
just in case: same as Efe's comment about wrapping the error with `nlb.additional_listener[%d]` instead of `nlb.listener[%d]`!